crates/sui-core/src/checkpoints/causal_order_effects.rs

```diff
@@ -192,10 +192,11 @@ impl EffectsStore for Arc<AuthorityStore> {
     }
 }
 
-/// An identity causal order that returns just the same order. For testing.
-pub struct TestCausalOrderNoop;
+/// A transaction effects store that returns an identity causal order. For testing.
+#[derive(Default)]
+pub struct TestEffectsStore(pub BTreeMap<TransactionDigest, TransactionEffects>);
 
-impl CausalOrder for TestCausalOrderNoop {
+impl CausalOrder for TestEffectsStore {
```

Comment:
I think you can implement CausalOrder for BTreeMap directly, without the newtype.

Reply:
We can but I think using a newtype here is more clear and provides better abstraction than directly using BTreeMap. There's also no runtime overhead afaik since the compiler optimizes it away to just the inner type, if that's what you are concerned about. I don't have strong preference though.

Reply:
Oh I don't have a preference, this looks clean.